library/src/main/scala/scala/scalajs/js/Tuple.scala

```diff
@@ -0,0 +1,695 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js API               **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-lang.org/     **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+/**
+ * All doc-comments marked as "MDN" are by Mozilla Contributors,
+ * distributed under the Creative Commons Attribution-ShareAlike license from
+ * https://developer.mozilla.org/en-US/docs/Web/Reference/API
+ */
+package scala.scalajs.js
+
+import scala.scalajs.js.annotation.JSName
+import scala.language.implicitConversions
+
+
+// scalastyle:off line.size.limit
+
+/**
+ *  A tuple "view" of 2 elements of a JavaScript [[Array]].
+ *  Combines
+ *  {{{
+ *  0: T0;
+ *  1: T1;
+ *  }}}
+ *  to
+ *  {{{
+ *  js.Tuple2[T0,T1]
+ *  }}}
+ *
+ *  Supports implicit conversion to [[scala.Tuple2]].
+ *  To use it, cast your array into a [[Tuple2]] using
+ *  {{{
+ *  val array = js.Array[Any](42, "foobar")
+ *  val tuple2 = array.asInstanceOf[js.Tuple2[Int, String]]
+ *  }}}
+ *  or convert a Scala tuple
+ *  {{{
+ *  val obj: js.Tuple2[Int, String] = (42, "foobar")
+ *  }}}
+ *  @see [[Tuple3]]
+ *  @see [[Tuple4]]
+ *  @see [[Tuple5]]
+ *  @see [[Tuple6]]
+ *  @see [[Tuple7]]
+ *  @see [[Tuple8]]
+ *  @see [[Tuple9]]
+ *  @see [[Tuple10]]
+ *  @see [[Tuple12]]
+ *  @see [[Tuple13]]
+ *  @see [[Tuple14]]
+ *  @see [[Tuple15]]
+ *  @see [[Tuple16]]
+ *  @see [[Tuple17]]
+ *  @see [[Tuple18]]
+ *  @see [[Tuple19]]
+ *  @see [[Tuple20]]
+ *  @see [[Tuple21]]
+ *  @see [[Tuple22]]
```

Comment:
Hum ... My idea was to put `@see [[Tuple2]]` from the other tuple traits, so the other way around. So that if you come to the `js.Tuple4` trait doc, you're sent to the `js.Tuple2` page which actually contains some doc.